src/transformations.jl

```diff
@@ -270,45 +266,22 @@ function fit(::Type{UnitRangeTransform}, X::AbstractMatrix{<:Real};
         Base.depwarn("fit(t, x) is deprecated: use fit(t, x, dims=2) instead", :fit)
         dims = 2
     end
-    if dims == 1
-        l, tmin, tmax = _compute_extrema(X)
-    elseif dims == 2
-        l, tmin, tmax = _compute_extrema(X')
-    else
-        throw(DomainError(dims, "fit only accept dims to be 1 or 2."))
-    end
-
-    for i = 1:l
-        @inbounds tmax[i] = 1 / (tmax[i] - tmin[i])
-    end
+    dims ∈ [1,2] || throw(DomainError(dims, "fit only accept dims to be 1 or 2."))
+    tmin_tmax = extrema(X; dims=dims)
+    tmin, tmax = (vec(getindex.(tmin_tmax, i)) for i in 1:2)
```

Comment:
Ah so it's too bad that this forces copying the result to extract the min and max. Do you think CUDA supports `eachslice`, or would it be efficient to iterate over rows/columns and call `extrema` on them? Then we could store directly the result in the right format.

Reply:
I have been experimenting a little bit, but have not found a good solution. This prevents the copying, but does not work with `CuArray`s because of the element-wise setindex:
```julia
l = size(X, dims==1 ? 2 : 1)
tmin, tmax = (similar(X, l) for _ in 1:2)
for (i, x) in enumerate(eachslice(X; dims=dims))
    tmin[i], tmax[i] = extrema(x) # per-element assignment does not work with CUDA
end
```
Then there is `zip` + `...` but I think the performance of that is even worse because of the additional penalty of `...`:
```julia
tmin, tmax = collect.(zip((extrema(x) for x in eachslice(X; dims=dims))...))
```
Also, now the outputs no longer end up in a `CuArray` which will mean the `transform!` will fail on `CuArray`.

Reply:
Ah, too bad. Maybe better not implement this until CUDA supports it. Could you ask for advice there about the best way to achieve this without copying?

Reply:
I asked the question on discourse, see [here](https://discourse.julialang.org/t/efficient-and-cuarray-compatible-implementation-of-statsbase-compute-extrema/51671).

Reply:
Might also be worth posting a link on #gpu on Slack.

Reply:
In the latest commit I added a new  `_compute_extrema` that eliminates the allocation. With that, it should now be as efficient as the original implementation for regular arrays. Also, this allow for a temporary solution for `CuArray`s by implementing as specialized `_compute_extrema` for `CuArray`s:
```julia
function  StatsBase._compute_extrema(X::CuArray; dims)
    tmin = vec(minimum(X; dims=dims))
    tmax = vec(maximum(X; dims=dims))
    return tmin, tmax
end
```
This is not optimal, but it works. Also, given the compute power of a GPU, I assume it will still give a significant performance improvement.

Reply:
Cool. Though you'll need Compat 2.2 to use `eachslice` on Julia 1.0.